public/app/features/chat/ChatMessage.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FunctionComponent, useState } from 'react';
+import { IconButton, MenuGroup, MenuItem, WithContextMenu } from '@grafana/ui';
+import { sanitize } from '@grafana/data/src/text/sanitize';
+import { marked } from 'marked';
+import { ChatMessageAction } from './ChatMessageAction';
+import { User } from './ChatUser';
+
+export interface Message {
+  id: number;
+  content: string;
+  created: number;
+  userId: number;
+  user: User;
+}
+
+function renderChatMarkdown(str?: string): string {
+  const html = marked(str || '', {
+    pedantic: false,
+    gfm: true,
+    smartLists: true,
+    smartypants: false,
+    xhtml: false,
+    breaks: true,
+  });
+  return sanitize(html);
+}
+
+interface ChatMessageProps {
+  message: Message;
+  actions?: ChatMessageAction[];
+}
+
+export const ChatMessage: FunctionComponent<ChatMessageProps> = ({ message, actions = [] }) => {
```

Comment:
You do not need to use `: FunctionComponent<ChatMessageProps> ` typing. It's not considered good practice anymore. Just write 

ts
```
export function ChatMessage({ message, actions = [] }: Props) { 
}
```

instead and rely on implicit typing